### PR TITLE
Backend

### PR DESCRIPTION
**Pull request by Nat3005**
Pull request for merging backend branch to main. Backend branch includes environment for backend app (Python, Flask, and additional libraries (Black, PyTest) made using Poetry). Docker directory consists of Dockerfile.backend.

### CODE REVIEW

**Comment by mwidera** [comment]
Looks almost good :) 

Things that possibly could be improved:
1. Please execute formatter (black) for python files
2. Please add `.gitignore` file as it might help to sort out cached files (e.g. `__pycache__`)
3. This is a minor one: version in `pyproject.toml` for python should be the same, as the dockerfile defined one (right now we expect at least 3.9 (range between 3.9.0 to 4.0.0)
4.  Dockerfile for backend:
  a. poetry should be installed via cmd `curl -sSL https://install.python-poetry.org | python -` 
  [installation via pip is not recommended as it may interfere with other packages](https://python-poetry.org/docs/#installing-with-pip)

File: backend/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[tool.poetry]
+name = "backend"
+version = "0.1.0"
+description = "Backend for loancalculator"
+authors = ["Your Name <you@example.com>"]
+
+[tool.poetry.dependencies]
+python = "^3.9"
```
Comment (mwidera) [inline backend/pyproject.toml:8]: Please align to version defined in backend Dockerfile

File: docker/Dockerfile.backend [hunk 2]
```diff
@@ -0,0 +1,15 @@
+FROM python:3.9.10-buster
+
+ENV PIP_DISABLE_PIP_VERSION_CHECK=on
+
+RUN pip install poetry
```
Comment (mwidera) [inline docker/Dockerfile.backend:5]: Please replace with `curl` suggested by poetry developers

File: docker/Dockerfile.backend [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+WORKDIR /app
+COPY backend/poetry.lock backend/pyproject.toml /app/
+
+RUN poetry config virtualenvs.create false
+RUN poetry install --no-interaction
```
Comment (mwidera) [inline docker/Dockerfile.backend:11]: Reduce `RUN` via multi line call

File: backend/app.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+app = Flask(__name__)
+
+# import requests
+# page = requests.get("http://www.example.com")
+# from bs4 import BeautifulSoup
+# soup = BeautifulSoup(page.content, 'html.parser')
+
+@app.route('/')
+def hello_world():
+    return r'<!DOCTYPE html><html lang="en"><head><meta charset = "UTF-8"><meta name="viewport" content="width=device-width, initial-scale=1.0">  <title>Hello Front-End!</title></head><body><h4 style="color: #001135;position: fixed;top: 50%; left: 50%;transform: translate(-50%, -50%);">Hello front-end!</h4></body></html>'
```
Comment (mwidera) [inline backend/app.py:11]: `black` formatter should make it more readable, but I hope this route will disappear soon

File: docker/Dockerfile.backend [hunk 5]
```diff
@@ -0,0 +1,15 @@
+FROM python:3.9.10-buster
+
+ENV PIP_DISABLE_PIP_VERSION_CHECK=on
+
+RUN pip install poetry
+
+WORKDIR /app
+COPY backend/poetry.lock backend/pyproject.toml /app/
```
Comment (kkuriata) [inline docker/Dockerfile.backend:8]: Maybe just `COPY` everything at once

**Review by kkuriata — CHANGES_REQUESTED** [review]

**Comment by Sakue34** [comment]
Fixed python version mismatch in pyproject.toml, added authors, changed poetry installation method to recommended one and updated Dockerfile.backend according to requested changes.

File: backend/app.py (hunk 4 above)
Comment (Sakue34) [reply backend/app.py]: Done

File: backend/app.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+from flask import Flask
+app = Flask(__name__)
+
+# import requests
```
[the remark this answers is not in the document]
Comment (Sakue34) [reply backend/app.py]: Done

File: backend/pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+Flask = "^2.0.3"
+beautifulsoup4 = "^4.10.0"
+black = "^22.1.0"
```
[the remark this answers is not in the document]
Comment (Sakue34) [reply backend/pyproject.toml]: Changed to dev dependencies

File: docker/Dockerfile.backend (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sakue34) [reply docker/Dockerfile.backend]: Done

File: docker/Dockerfile.backend (hunk 5 above)
Comment (Sakue34) [reply docker/Dockerfile.backend]: Done

**Review by kkuriata — APPROVED** [review]

**Review by mwidera — APPROVED** [review]

**Pull request merged**